gt/lua_lpm.c

```diff
@@ -161,9 +167,9 @@ l_lpm_add(lua_State *l)
 {
 	int ret;
 
-	/* First argument must be of type struct rte_lpm **. */
-	struct rte_lpm *lpm =
-		*(struct rte_lpm **)luaL_checkudata(l, 1, LUA_LPM_TNAME);
+	/* First argument must be of type struct lpm_lua_userdata *. */
+	struct lpm_lua_userdata *lpm_ud =
+		(struct lpm_lua_userdata *)luaL_checkudata(l, 1, LUA_LPM_UD_TNAME);
```

Comment:
Since we no longer expect a double-redirection pointer from `luaL_checkudata()`, the typecast may no longer be needed. Could you drop `(struct lpm_lua_userdata *)`? If so, please carefully review the other calls of `luaL_checkudata()` to drop the typecast.

This comment applies to the IPv6 cases below as well.